provider/cloudflare/cloudflare.go

```diff
@@ -213,7 +229,7 @@ func (p *CloudFlareProvider) Records(ctx context.Context) ([]*endpoint.Endpoint,
 
 	endpoints := []*endpoint.Endpoint{}
 	for _, zone := range zones {
-		records, err := p.Client.DNSRecords(ctx, zone.ID, cloudflare.DNSRecord{})
+		records, _, err := p.Client.ListDNSRecords(ctx, cloudflare.ZoneIdentifier(zone.ID), cloudflare.ListDNSRecordsParams{})
```

Comment:
I don't really mind but it's interesting that the interface function changes the result signature which is not used, so why change it?

Reply:
I think the original intent on cloudflare-go was to return the `ResultInfo` struct which allows the operator to move between pages: https://github.com/cloudflare/cloudflare-go/pull/1151/files#diff-cb1b8a01bbc5f64cf06dce0e3f8021ad8289078b402dc620afa04780484d7cd5R157

Since we're using the implicit auto pagination feature (that was already present) we don't really need to use this struct here.

Reply:
thanks!